independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanDeployment.java

```diff
@@ -638,9 +644,21 @@ static ScopeInfo getValidScope(Collection<ScopeInfo> stereotypeScopes, Annotatio
                                 beanClass);
                         beanClasses.add(beanClass);
                     }
-                } else if (annotationStore.hasAnnotation(method, DotNames.DISPOSES)) {
-                    // Disposers are not inherited
-                    disposerMethods.add(method);
+                } else {
+                    // Verify that non-producer methods are not annotated with stereotypes
+                    // only account for 'real' stereotypes that are not additional BeanDefiningAnnotations
+                    for (AnnotationInstance i : annotationStore.getAnnotations(method)) {
```

Comment:
I'd move this logic here: https://github.com/quarkusio/quarkus/blob/master/independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanDeployment.java#L660 (to verify the inherited methods as well).